### ISSUE TITLE
Dokan virtual disk not creating on Win7 x64

### ISSUE THREAD

**Issue by Xardion495** [jetwhiz/encfs4win#120]
# Environment
 * Windows version: Windows 7
 * Processor architecture: x64
 * Encfs4win version: 1.10.1-RC13
 * Dokan version (legacy, modern): modern 1.1.0.2000

# Description 
After the mounting crypto container nothing happen. Dokan virtual disk not creating.
Early it was working on the previous exemplar of OS Win7 x64 but not on this one.
## Expected behavior vs. actual behavior 

Expected the working utility

## Steps to reproduce problem
1. Place enc_key (.encfs6.xml) to the target directory;
2. Boot the encfsw.exe with admin previligies;
3. Mount the crypto container with entering password.

# Relevant logs

Unavailable.

**Comment by jetwhiz** [comment]
Thank you for the bug report, @Xardion495!  

We have released RC14, based on Dokany 1.2.1.2000.  Can you test on this version and see if this fixes your issue? 

If not, can you mount your directory using encfs.exe instead of via the GUI?   For example: 
`encfs.exe -f C:\My\Folder\ Z:`

**Comment by Xardion495** [comment]
I reinstalled the app to the RC14. And tried mounting crypto container via the GUI and CMD (through the encfs.exe). Unfortunately, that didn't help.

**Comment by jetwhiz** [comment]
Everything seems to be working on my Windows 7 build machine, so this might be related to Dokany not being upgraded properly.  Can you check the Dokan program files folder and make sure that 'Dokan Library-1.2.1' is really installed?  

You might need to uninstall the older Dokany, restart and then install Dokany 1.2.1.2000 cleanly.

**Comment by Xardion495** [comment]
I checked all. The components have been installed correctly. I reinstalled their and restarted the machine - the result was the same.
I watched via ProcessExplorer the following:
1. When i make a mounting try, encfsw.exe calling encfs.exe
2. At the same moment csrss.exe generate conhost.exe
3. After about 10 seconds both of them are closed simultaneously.

**Comment by jetwhiz** [comment]
Can you mount the filesystem using encfs.exe command line instead of using the encfsw GUI?  If you use the "-v" verbose flag then it should give detailed output that can give us an idea as to why this is not mounting.  Any logs created should be available in your %TEMP% folder.

**Comment by Xardion495** [comment]
> 
> 
> Can you mount the filesystem using encfs.exe command line instead of using the encfsw GUI? If you use the "-v" verbose flag then it should give detailed output that can give us an idea as to why this is not mounting. Any logs created should be available in your %TEMP% folder.

There is a debug info:

C:\Program Files (x86)\encfs>encfs.exe -f -v D:\ButuzovDP_DATA\Encrypted_data Z:

2019-03-04 10:08:26,505 VER [main.cpp:697] Root directory: D:\ButuzovDP_DATA\Encrypted_data/
2019-03-04 10:08:26,510 VER [main.cpp:698] Fuse arguments: (fg) (encfs) (threaded) (keyCheck) encfs.exe Z: -f -f -o use_ino -o default_permissions
2019-03-04 10:08:26,593 VER [FileUtils.cpp:296] found new serialization format
2019-03-04 10:08:26,598 VER [FileUtils.cpp:310] subVersion = 20100713
2019-03-04 10:08:26,601 VER [Interface.cpp:110] checking if ssl/aes(3:0:2) implements ssl/aes(3:0)
2019-03-04 10:08:26,605 VER [SSL_Cipher.cpp:328] allocated cipher ssl/aes, keySize 32, ivlength 16
2019-03-04 10:08:26,610 VER [FileUtils.cpp:1613] useStdin: 0
EncFS Password:
2019-03-04 10:08:47,531 VER [Interface.cpp:110] checking if ssl/aes(3:0:2) implements ssl/aes(3:0)
2019-03-04 10:08:47,537 VER [SSL_Cipher.cpp:328] allocated cipher ssl/aes, keySize 32, ivlength 16
2019-03-04 10:08:48,157 VER [FileUtils.cpp:1622] cipher key size = 52
2019-03-04 10:08:48,162 VER [Interface.cpp:110] checking if nameio/block(4:0:2) implements nameio/block(4:0)
2019-03-04 10:08:48,209 VER [MACFileIO.cpp:70] fs block size = 1024, macBytes = 8, randBytes = 0
2019-03-04 10:08:48,215 VER [DirNode.cpp:667] created FileNode for D:\ButuzovDP_DATA\Encrypted_data/
2019-03-04 10:08:48,220 VER [encfs.cpp:129] op: getattr : D:\ButuzovDP_DATA\Encrypted_data/
2019-03-04 10:09:05,483 VER [openssl.cpp:47] Allocating 41 locks for OpenSSL

C:\Program Files (x86)\encfs>

**Comment by jetwhiz** [comment]
@Xardion495 -- this may be related to a known Dokany FUSE library issue. 

Can you try using the updated version: [v1.10.1-RC15](https://github.com/jetwhiz/encfs4win/releases/tag/v1.10.1-RC15).  Make sure you uninstall both the old encfs4win and old Dokany before installing RC15 (to be sure that Dokany properly upgrades).

**Comment by Xardion495** [comment]
> 
> 
> @Xardion495 -- this may be related to a known Dokany FUSE library issue.
> 
> Can you try using the updated version: [v1.10.1-RC15](https://github.com/jetwhiz/encfs4win/releases/tag/v1.10.1-RC15). Make sure you uninstall both the old encfs4win and old Dokany before installing RC15 (to be sure that Dokany properly upgrades).

Done.
The result is still the same.

C:\Program Files (x86)\encfs>encfs.exe -f -v D:\ButuzovDP_DATA\Encrypted_data G:
2019-04-22 11:27:53,548 VER [main.cpp:697] Root directory: D:\ButuzovDP_DATA\Encrypted_data/
2019-04-22 11:27:53,553 VER [main.cpp:698] Fuse arguments: (fg) (encfs) (threaded) (keyCheck) encfs.exe G: -f -f -o use_ino -o default_permissions
2019-04-22 11:27:53,632 VER [FileUtils.cpp:296] found new serialization format
2019-04-22 11:27:53,637 VER [FileUtils.cpp:310] subVersion = 20100713
2019-04-22 11:27:53,641 VER [Interface.cpp:110] checking if ssl/aes(3:0:2) implements ssl/aes(3:0)
2019-04-22 11:27:53,647 VER [SSL_Cipher.cpp:328] allocated cipher ssl/aes, keySize 32, ivlength 16
2019-04-22 11:27:53,652 VER [FileUtils.cpp:1613] useStdin: 0
EncFS Password:
2019-04-22 11:28:09,166 VER [Interface.cpp:110] checking if ssl/aes(3:0:2) implements ssl/aes(3:0)
2019-04-22 11:28:09,173 VER [SSL_Cipher.cpp:328] allocated cipher ssl/aes, keySize 32, ivlength 16
2019-04-22 11:28:09,777 VER [FileUtils.cpp:1622] cipher key size = 52
2019-04-22 11:28:09,782 VER [Interface.cpp:110] checking if nameio/block(4:0:2) implements nameio/block(4:0)
2019-04-22 11:28:09,829 VER [MACFileIO.cpp:70] fs block size = 1024, macBytes = 8, randBytes = 0
2019-04-22 11:28:09,835 VER [DirNode.cpp:677] created FileNode for D:\ButuzovDP_DATA\Encrypted_data/
2019-04-22 11:28:09,840 VER [encfs.cpp:129] op: getattr : D:\ButuzovDP_DATA\Encrypted_data/
2019-04-22 11:28:28,642 VER [openssl.cpp:47] Allocating 41 locks for OpenSSL

**Comment by jetwhiz** [comment]
I haven't had any luck recreating this issue yet with any of my build machines (Windows 7, Windows 8.1 and Windows 10).  If it is automatically disconnecting itself, then this may be an issue with the Dokan FUSE install.  

Can you try the Dokany `fuse_mirror` sample and see if the problem exists there, too?

It is located in "C:\Program Files\Dokan\Dokan Library-1.x.x\sample\fuse_mirror", and you need Cygwin to be installed to run it.

You can mount the default directory just by running `mirror.exe Z:`, for example.

**Comment by Xardion495** [comment]
There are some difficulties with running mirror.exe through the cygwin's mintty.exe
I tried to run the utility as administrator, added a few folders to PATH-environment and etc. but it had no effect.

Unfortunately this issue took too much time and i a little tired from all of that. I came to the conclusion that i have a some kind of specific circumstance in my case preventing the correct operation of the utility on my machine. This applies to the workstation only. On my home PC it working correctly.

So i closing issue. Thank you for your attempt to help with the solution.

**Comment by raven428** [comment]
@jetwhiz I got same issue (with same output from verbose mode) repeating on multiple virtual and physical machines deployed from same preconfigured image. I suggest there is some issue in the image configuration, but wasn't found this issue. Empty installed Windows works fine with `encfs4win`.

I tried run `mirror.exe z:` from `C:\Program Files\Dokan\Dokan Library-1.x.x\sample\fuse_mirror` on machine where `encfs4win` couldn't mount anything. It return empty line to console and also do not mount anything. On empty Windows installation it also return empty line, but mount cygwin root into `z:` disk.

Are there any more verbose logs to discover root of this issue?

Update: This repeats only in terminal session session in both windows 7 and 10. When running `encfs.exe -v -f folder s:` in physical session it's working fine. When runs this command inside terminal session with administrative rights (Run as Administrator) it also working fine.

Also, I tried next sequence:
1. `encfs.exe -v -f folder s:` in physical session without administrative rights - it working fine.
2. Without closing the session, I'm connecting to this session thru terminal services (switching session from physical to terminal).
3. After successful verbose records at the console I'm getting `2020-06-28 18:24:36,533 VER [openssl.cpp:47] Allocating 41 locks for OpenSSL` and virtual disk became dismounted then disappeared in Windows 10 and stayed unavailable in Windows 7.

So, the final question: is there a way to make ability to use `encfs4win` in terminal session w/o administrative privileges?
